src/main/java/org/apache/maven/shared/artifact/filter/PatternIncludesArtifactFilter.java

```diff
@@ -107,322 +147,783 @@ public boolean include( final Artifact artifact )
     }
 
     /**
-     * @param artifact check for positive match.
-     * @return true/false.
+     * @return description.
      */
-    protected Boolean positiveMatch( final Artifact artifact )
+    protected String getFilterDescription()
     {
-        if ( positivePatterns == null || positivePatterns.isEmpty() )
+        return "artifact inclusion filter";
+    }
+
+    /** {@inheritDoc} */
+    public void reportFilteredArtifacts( final Logger logger )
+    {
+        if ( !filteredArtifact.isEmpty() && logger.isDebugEnabled() )
         {
-            return null;
+            final StringBuilder buffer =
+                new StringBuilder( "The following artifacts were removed by this " + getFilterDescription() + ": " );
+
+            for ( Artifact artifactId : filteredArtifact )
+            {
+                buffer.append( '\n' ).append( artifactId.getId() );
+            }
+
+            logger.debug( buffer.toString() );
         }
-        else
+    }
+
+    /** {@inheritDoc} */
+    public boolean hasMissedCriteria()
+    {
+        // if there are no patterns, there is nothing to report.
+        if ( !patterns.isEmpty() )
         {
-            return match( artifact, positivePatterns );
+            final List<Pattern> missed = new ArrayList<>( patterns );
+            missed.removeAll( patternsTriggered );
+
+            return !missed.isEmpty();
         }
+
+        return false;
     }
 
-    private boolean match( final Artifact artifact, final List<String> patterns )
+    private static final char[] EMPTY = new char[0];
+
+    private static final char[] ANY = new char[] { '*' };
+
+    static char[] emptyOrChars( String str )
     {
-        final String shortId = ArtifactUtils.versionlessKey( artifact );
-        final String id = artifact.getDependencyConflictId();
-        final String wholeId = artifact.getId();
+        return str != null && str.length() > 0 ? str.toCharArray() : EMPTY;
+    }
 
-        if ( matchAgainst( wholeId, patterns, false ) )
-        {
-            return true;
-        }
+    static char[] anyOrChars( char[] str )
+    {
+        return str.length > 1 || ( str.length == 1 && str[0] != '*' ) ? str : ANY;
+    }
 
-        if ( matchAgainst( id, patterns, false ) )
+    static char[][] tokenizeAndSplit( String pattern )
+    {
+        String[] stokens = pattern.split( ":" );
+        char[][] tokens = new char[ stokens.length ][];
+        for ( int i = 0; i < stokens.length; i++ )
         {
-            return true;
+            tokens[i] = emptyOrChars( stokens[i] );
         }
+        return tokens;
+    }
 
-        if ( matchAgainst( shortId, patterns, false ) )
+    static boolean match( char[] patArr, char[] strArr, boolean isVersion )
+    {
+        int patIdxStart = 0;
+        int patIdxEnd = patArr.length - 1;
+        int strIdxStart = 0;
+        int strIdxEnd = strArr.length - 1;
+        char ch;
+
+        boolean containsStar = false;
```

Comment:
I think the term is `wildcard` instead of star.

Reply:
Though, in this case, we specifically look for a '*'.  Note that given the `static boolean match( char[] patArr, char[] strArr, boolean isVersion )` has been borrowed from the plexus-utils `SelectorUtils`, it also supports the `?` wildcard. That should not be a problem since it's not a valid character in any of the coordinates.  But this variable is really about looking for the star and not the question mark, so I'd be leaning to keep the name. 
Though, I'm not opposed to renaming if you still want.